packages/api/src/models/webhook.ts

```diff
@@ -3,6 +3,17 @@ import mongoose, { Schema } from 'mongoose';
 
 export enum WebhookService {
   Slack = 'slack',
+  Generic = 'generic',
+}
+
+interface MongooseMap extends Map<string, string> {
+  // https://mongoosejs.com/docs/api/map.html#MongooseMap.prototype.toJSON()
+  // Converts this map to a native JavaScript Map for JSON.stringify(). Set the flattenMaps option to convert this map to a POJO instead.
+  // doc.myMap.toJSON() instanceof Map; // true
+  // doc.myMap.toJSON({ flattenMaps: true }) instanceof Map; // false
+  toJSON: (options?: {
```

Comment:
so we need to set `flattenMaps` to true to convert the document to JSON ? not very intuitive

Reply:
Naw, need to set flattenMaps to true only if the input has nested Maps - in these cases it's probably not needed at all since inputting nested maps would rarely happen. the regular call to toJSON() it still preps it for use with json.stringify etc.

Reply:
I didn't need them for queryParams/headers I should have left those out, nobody is inputting raw maps their either, that was my mistake

Reply:
we really just want to make sure the object is valid for JSON.stringify() and the raw method does that 
![image](https://github.com/hyperdxio/hyperdx/assets/4743932/88efb395-18f1-4d6a-a1fb-30f38ae175bd)

Reply:
I see. Thanks for the detailed explanation. This makes sense.